src/v2/Components/FollowButton/FollowArtistButton.tsx

```diff
@@ -169,66 +157,69 @@ export class FollowArtistButton extends React.Component<Props, State> {
       },
       variables: {
         input: {
-          // @ts-expect-error STRICT_NULL_CHECK
           artistID: artist.internalID,
-          // @ts-expect-error STRICT_NULL_CHECK
           unfollow: artist.is_followed,
         },
       },
     })
     this.trackFollow()
-    // @ts-expect-error STRICT_NULL_CHECK
-    if (triggerSuggestions && !artist.is_followed) {
-      this.setState({ openSuggestions: true })
-    }
-  }
-
-  closePopover() {
-    this.setState({ openSuggestions: false })
   }
 
   render() {
-    const { artist, buttonProps, render, user } = this.props
-    const { openSuggestions } = this.state
-
-    // Custom button renderer
-    const content = render ? (
-      <Container
-        tabIndex={0}
-        role="button"
-        onKeyPress={event => {
-          if (event.key === "Enter" || event.key === " ") {
-            this.handleFollow(event)
-          }
-        }}
-        onClick={this.handleFollow}
-      >
-        {" "}
-        {/* @ts-expect-error STRICT_NULL_CHECK */}
-        {render(artist)}
-      </Container>
-    ) : (
-      <FollowButton
-        // @ts-expect-error STRICT_NULL_CHECK
-        isFollowed={artist && artist.is_followed}
-        handleFollow={this.handleFollow}
-        buttonProps={buttonProps}
-      />
-    )
+    const { artist, buttonProps, render, user, triggerSuggestions } = this.props
 
     return (
       <Box data-test="followArtistButton">
-        {content}
-        {openSuggestions && (
-          <SuggestionsPopoverContainer>
-            <SuggestionsPopover
+        <Popover
+          title="Other artists you might like"
+          placement="bottom"
+          popover={
+            <FollowArtistPopoverFragmentContainer
               user={user}
-              // @ts-expect-error STRICT_NULL_CHECK
-              artist={artist}
-              onClose={() => this.closePopover()}
+              artist={artist!}
             />
-          </SuggestionsPopoverContainer>
-        )}
+          }
+        >
+          {({ anchorRef, onVisible }) => {
+            const openSuggestions = () => {
+              if (triggerSuggestions && !artist?.is_followed) {
+                onVisible()
```

Comment:
what does `onVisible` do?

Reply:
Opens the popover

Reply:
I see, so you can set `<Popover visible /> or use this form to display 👍

Reply:
Exactly